File: src/lib/api/playback.ts
// Playback history (Phase 4 — SQLite-backed).
import { invoke } from "@tauri-apps/api/core";

export interface PlaybackHistoryEntry {
  file_path: string;
  series_key: string | null;
  episode: number | null;
  title: string | null;
  position: number;
  duration: number;
}

/** Save or update the playback position for a file. */
export async function savePlaybackPosition(
  entry: PlaybackHistoryEntry,
): Promise<void> {
  return invoke("save_playback_position", { entry });
}

/** Get the saved playback position (in seconds) for a file. */
export async function getPlaybackPosition(
  filePath: string,
): Promise<number | null> {
  return invoke<number | null>("get_playback_position", { filePath });
}

/** Delete the playback history entry for a file (e.g. after 85%+ watched). */
export async function deletePlaybackPosition(filePath: string): Promise<void> {
  return invoke("delete_playback_position", { filePath });
}

/** Get recent playback entries for "Continue Watching" UI. */
export async function recentPlayback(): Promise<PlaybackHistoryEntry[]> {
  return invoke<PlaybackHistoryEntry[]>("recent_playback");
}

File: src-tauri/src/commands/playback.rs
//! Playback history Tauri commands.

use tauri::State;
use crate::db::{self, Db, PlaybackHistoryEntry};

/// Save or update the playback position for a file.
#[tauri::command]
pub async fn save_playback_position(
    entry: PlaybackHistoryEntry,
    db: State<'_, Db>,
) -> Result<(), String> {
    db::save_playback_position(&db.0, &entry).await
}

/// Get the saved playback position (in seconds) for a file.
#[tauri::command]
pub async fn get_playback_position(
    file_path: String,
    db: State<'_, Db>,
) -> Result<Option<f64>, String> {
    db::get_playback_position(&db.0, &file_path).await
}

/// Delete the playback history entry for a file.
#[tauri::command]
pub async fn delete_playback_position(
    file_path: String,
    db: State<'_, Db>,
) -> Result<(), String> {
    db::delete_playback_position(&db.0, &file_path).await
}

/// Get recent playback entries for "Continue Watching" UI.
#[tauri::command]
pub async fn recent_playback(
    db: State<'_, Db>,
) -> Result<Vec<PlaybackHistoryEntry>, String> {
    db::recent_playback(&db.0, 20).await
}
